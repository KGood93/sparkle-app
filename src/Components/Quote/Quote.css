.motivQuote {
    border: 3px solid black;
    margin: 15px;
    margin-bottom: 0px;
    width: 350px;
}

.motivQuote h2 {
    margin: 5px;
    text-align: center;
}

.motivQuote h4 {
    margin: 5px;
    text-align: center;
}