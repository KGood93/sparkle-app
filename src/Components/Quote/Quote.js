import React, {Component} from 'react';
import './Quote.css';
import {findQuote} from '../../entry-helpers'
import ApiContext from '../../ApiContext'

class Quote extends Component {
    static contextType = ApiContext

    render() {
      const {quotes = []} = this.context
      //console.log(quotes)
      const quoteid = this.props.quoteid
      console.log(quoteid)
      const quoteId = parseInt(quoteid)
      //console.log(quoteId)
      const disQuote = findQuote(quotes, quoteId) || {quote: '', author: '', used: ''}
      console.log(disQuote)
        return (
            <div className="entryMain">
              <section className="motivQuote">
                <h2>{disQuote.quote}</h2>
                <h4 className="quoteBy">{disQuote.author}</h4>
              </section>
            </div>
        )
    }
}
export default Quote;