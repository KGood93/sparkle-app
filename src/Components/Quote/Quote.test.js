import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Quote from './Quote';

describe('Quote Component', () => {
    it(`renders without crashing`, () => {
        const wrapper = shallow(<Quote />);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});