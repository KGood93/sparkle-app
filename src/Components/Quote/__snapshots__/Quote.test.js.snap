// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quote Component renders without crashing 1`] = `
<div
  className="entryMain"
>
  <section
    className="motivQuote"
  >
    <h2 />
    <h4
      className="quoteBy"
    />
  </section>
</div>
`;
